src/components/datagrid/datagrid.tsx

```diff
@@ -0,0 +1,227 @@
+import { Paginator, PaginatorProps } from "../paginator";
+import { Toolbar } from "../toolbar";
+import { A, AProps, H3, P } from "../typography";
+import "./datagrid.scss";
+
+/** Matches a URL. */
+const REGEX_URL = /https?:\/\/[^\s]+$/;
+
+export type RowData = Record<string, boolean | number | string | null>;
+
+export type DataGridProps = Omit<
+  React.HTMLAttributes<HTMLDivElement>,
+  "results"
+> & {
+  /** The results (after pagination), only primitive types supported for now. */
+  results: RowData[];
+
+  /** A `string[]` containing the keys in `results` to show data for. */
+  fields?: string[];
+
+  /**
+   *  A `string[]` containing the fields which should be used to detect the url
+   *  of a row. Fields specified in this array won't be rendered, instead: the
+   *  first (non url) field is rendered as link (`A`) with `href` set to the
+   *  resolved url of the row.
+   */
+  urlFields?: string[];
+
+  /** Props for A. */
+  aProps?: Omit<AProps, "href">;
+
+  /** Props for Badge. */
+  badgeProps?: BadgeProps;
+
+  /** Props for Boolean. */
+  booleanProps?: Omit<BooleanProps, "value">;
+
+  /** If set, the paginator is enabled. */
+  paginatorProps?: PaginatorProps;
+
+  /** A title for the datagrid. */
+  title?: string;
+};
+
+/**
+ * DataGrid component
+ * @param aProps
+ * @param badgeProps
+ * @param booleanProps
+ * @param paginatorProps
+ * @param results
+ * @param fields
+ * @param title
+ * @param urlFields
+ * @param props
+ * @constructor
+ */
+export const DataGrid: React.FC<DataGridProps> = ({
+  aProps,
+  badgeProps,
+  booleanProps,
+  results,
+  fields = results?.length ? Object.keys(results[0]) : [],
+  paginatorProps,
+  title = "",
+  urlFields = [
+    "absolute_url",
+    "get_absolute_url",
+    "href",
+    "get_href",
+    "url",
+    "get_url",
+  ],
+  ...props
+}) => {
+  const id = useId();
+  const renderableFields = fields.filter((f) => !urlFields.includes(f));
+  const captions = renderableFields.map((f) => field2Caption(f as string));
+  const titleId = title ? `${id}-caption` : undefined;
+
+  /**
+   * Renders a cell based on type of `rowData[field]`.
+   * @param rowData
+   * @param field
+   * @param index
+   */
+  const renderCell = (rowData: RowData, field: string, index: number) => {
+    const fieldIndex = renderableFields.indexOf(field);
+    const urlField = urlFields.find((f) => String(rowData[f]).match(REGEX_URL));
+    const rowUrl = urlField ? rowData[urlField] : null;
+    const data = rowData[field];
+    const type = typeof data;
+
+    // Explicit link: passed as URL without being set as urlField.
+    const isExplicitLink = String(data).match(REGEX_URL);
+
+    // Implicit link: first column and `rowUrl` resolved using `urlFields`.
+    const isImplicitLink = rowUrl && fieldIndex === 0;
+
+    // Cell should be a link is truthy.
+    const link = isExplicitLink
+      ? String(data)
+      : isImplicitLink
+        ? String(rowUrl)
+        : "";
+
```

Comment:
I understand why we are using the ternaries here, but it is very unreadable. Could we slightly refactor this for readability?